azure_li_services/units/user.py

```diff
@@ -97,7 +100,7 @@ def setup_ssh_authorization(user):
 
 
 def setup_sudo_authorization(user):
-    if 'shadow_hash' in user:
+    if 'shadow_hash' in user and user['username'] != 'root':
```

Comment:
I think you're saying here that if we have a password for root, you don't want to change the root password (I didn't answer your E-Mail on this yet).

To be clear: I completely agree with you. I would NEVER have a root password enabled system in production. But that said, I can't force user behavior. As my wife says in relation to my child (and his distaste for going to sleep at any sort of reasonable hour), "You can lead a horse to water, but you can't make them drink." We ask him go downstairs at a reasonable hour, but we can't force him to lie down and close his eyes. Very frustrating!

By the same token, we offer both passwords and SSH keys. We point out that SSH keys are more secure. But if the user really insists on a password, we can't force them.

Please allow root passwords as well as SSH keys; for us, that's a requirement. Azure behaves that way, and we should too.

Reply:
>I think you're saying here that if we have a password for root, you don't want to
>change the root >password 

no this is in the setup_sudo_authorization() method and root always has any rights, there is no reason to setup sudo for root

Reply:
of course you can set the root password for root see the code in create_user() method.
(I think I should change the name, as this method creates or modifies)